chemprop/models/model.py

```diff
@@ -157,38 +158,43 @@ def on_validation_model_eval(self) -> None:
         self.predictor.output_transform.train()
 
     def validation_step(self, batch: TrainingBatch, batch_idx: int = 0):
-        losses = self._evaluate_batch(batch)
-        metric2loss = {f"val/{m.alias}": l for m, l in zip(self.metrics, losses)}
+        self._evaluate_batch(batch, "val")
 
-        self.log_dict(metric2loss, batch_size=len(batch[0]))
+        bmg, V_d, X_d, targets, weights, lt_mask, gt_mask = batch
+
+        mask = targets.isfinite()
+        targets = targets.nan_to_num(nan=0.0)
+
+        Z = self.fingerprint(bmg, V_d, X_d)
+        preds = self.predictor.train_step(Z)
+        loss = self.metrics[-1](preds, targets, mask, weights, lt_mask, gt_mask)
```

Comment:
I see now this is a change that was supposed to go in the torchmetrics PR. In that PR, the last element in metrics is a copy of the criterion and not the criterion itself because the torchmetric has a saved state that can't be shared between training and validation. I could implement this change here, but would need to update the torchmetrics PR, so I would prefer no to.

Reply:
Let's do this in the torchmetrics PR.